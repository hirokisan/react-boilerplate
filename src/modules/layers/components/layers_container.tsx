import React from 'react';
import { styled, theme } from 'config/common';
import { LayersTabs } from './layers_tabs';
import { ToolsContainer } from './tools_container';

export const LayersWrapper = styled.div`
  width: 300px;
  max-height: 60vh;
  background: ${theme.paneBackground};
  background-color: ${theme.paneBackground};
  display: flex;
  flex-direction: column;
  label: LayerContainer;
`;

LayersWrapper.displayName = 'LayersContainer';

const tabs = ['LAYERS', 'INDICATORS'];

export const LayersContainer = () => {
  const [tab, setTab] = React.useState(0);
  const stub: any = () => {};
  return (
    <LayersWrapper>
      <LayersTabs tabs={tabs} currentTab={tab} setTab={setTab} />
      <ToolsContainer setTool={stub} />
    </LayersWrapper>
  );
};
