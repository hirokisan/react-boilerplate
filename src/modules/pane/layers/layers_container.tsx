import React from 'react';
import { LayerItem } from './layer_item';
import { PaneContainer } from '../pane_container';

export const LayersContainer = ({ groups, openComments }) => {
  const zoomToLayer = React.useCallback((layer: any) => {
    console.log(layer);
  }, []);

  return (
    <PaneContainer>
      <ul className="list">
        {groups[0] && (
          <LayerItem
            layers={groups[0] ? groups[0] : []}
            zoomToLayer={(layer: any) => zoomToLayer(layer)}
            openComments={(layer: any) => openComments(layer)}
            folderName="Advanced AI Layers"
          />
        )}
        {groups[1] && (
          <LayerItem
            layers={groups[1] ? groups[1] : []}
            zoomToLayer={(layer: any) => zoomToLayer(layer)}
            openComments={(layer: any) => openComments(layer)}
            folderName="Derived Layers"
          />
        )}
        {groups[2] && (
          <LayerItem
            layers={groups[2] ? groups[2] : []}
            zoomToLayer={(layer: any) => zoomToLayer(layer)}
            openComments={(layer: any) => openComments(layer)}
            folderName="Primary Layers"
          />
        )}
      </ul>
    </PaneContainer>
  );
};
