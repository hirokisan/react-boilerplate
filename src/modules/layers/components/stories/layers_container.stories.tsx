import React from 'react';

import { action } from '@storybook/addon-actions';
import { ToolsContainer } from '../tools_container';
import { LayersContainer } from '../layers_container';

import marked from 'marked';

const docs = `
The layer container is one of the core system components, allowing user to display and manipulate layers. 
`;

export default {
  title: 'Components|Layers/Container',
  parameters: {
    component: ToolsContainer,
    info: marked(docs)
  }
};

export const DefaultView = () => <LayersContainer />;

DefaultView.story = {
  name: 'default'
};
