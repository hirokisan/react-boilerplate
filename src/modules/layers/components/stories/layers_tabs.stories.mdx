import { action } from '@storybook/addon-actions';
import { Story, Meta, Props, Source } from '@storybook/addon-docs/blocks';
import { text } from '@storybook/addon-knobs';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';

<Meta 
  title='Components|Layers/Tabs' 
  component={LayersTabs} 
  decorators={[(storyFn) => <LayersWrapper>{storyFn()}</LayersWrapper>]} 
/>

# Layer Tabs

Layer tabs switch between different layer groups

### Example

<Story name="first selected">
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={action('setTab')}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
</Story>

### Props

<Props of={LayersTabs} />

## Stories

### Second Selected

<Story name="second selected">
  <LayersTabs
    currentTab={1}
    setTab={action('setTab')}
    tabs={['LAYERS', 'INDICATORS']}
  />
</Story>