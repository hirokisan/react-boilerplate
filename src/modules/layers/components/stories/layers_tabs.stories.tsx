import React from 'react';
import { action } from '@storybook/addon-actions';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { LayersWrapper } from '../layers_container';
import { Action } from 'config/common';

export default {
  title: 'Components|Layers/Tabs',
  parameters: {
    component: LayersTabs,
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <LayersWrapper>{storyFn()}</LayersWrapper>]
  }
};

const storyProps = {
  setTab: action('setTab') as Action<number>
};

export const DefaultView = ({ setTab } = storyProps) => (
  <LayersTabs
    currentTab={number('ActiveTab', 0)}
    setTab={setTab}
    tabs={array('Tabs', ['LAYERS', 'INDICATORS'])}
  />
);

DefaultView.story = {
  name: 'first selected'
};

export const SelectedView = ({ setTab } = storyProps) => (
  <LayersTabs currentTab={1} setTab={setTab} tabs={['LAYERS', 'INDICATORS']} />
);

SelectedView.story = {
  name: 'second selected'
};
