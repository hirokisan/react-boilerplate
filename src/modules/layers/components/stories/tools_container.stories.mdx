import { action } from '@storybook/addon-actions';
import { Story, Meta, Props, Source } from '@storybook/addon-docs/blocks';
import { text } from '@storybook/addon-knobs';
import { LayersTabs } from '../layers_tabs';
import { array, number } from '@storybook/addon-knobs';
import { ToolsContainer } from '../tools_container';
import { LayersWrapper } from '../layers_container';

<Meta 
  title='Components|Layers/Tools' 
  component={ToolsContainer} 
  decorators={[(storyFn) => <LayersWrapper>{storyFn()}</LayersWrapper>]} 
/>

# Tools

Set of tools to operate pane content (e.g. layers)

### Example

<Story name="first selected">
  <ToolsContainer setTool={action('setTool')} />
</Story>

### Props

<Props of={ToolsContainer} />


