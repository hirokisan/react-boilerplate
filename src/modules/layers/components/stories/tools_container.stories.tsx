import React from 'react';

import { action } from '@storybook/addon-actions';
import { ToolsContainer } from '../tools_container';
import { LayersWrapper } from '../layers_container';

import docs from './tools_container.docs.md';
import marked from 'marked';
import { Action } from 'config/common';

export default {
  title: 'Components|Layers/Tools',
  parameters: {
    component: ToolsContainer,
    info: marked(docs),
    // eslint-disable-next-line react/display-name
    decorators: [(storyFn: Function) => <LayersWrapper>{storyFn()}</LayersWrapper>]
  }
};

const setTool = action('setTool') as Action<string>;

export const DefaultView = (setToolHandler = setTool) => (
  <ToolsContainer setTool={setToolHandler} />
);

DefaultView.story = {
  name: 'default'
};
