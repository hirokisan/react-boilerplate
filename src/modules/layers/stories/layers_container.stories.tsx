import React from 'react';

import marked from 'marked';
import { LayersContainer } from '../layers_container';

const docs = `
The layer container is one of the core system components, allowing user to display and manipulate layers. 
`;

export default {
  title: 'Components|Layers/Container',
  parameters: {
    component: LayersContainer,
    info: marked(docs)
  }
};

export const DefaultView = () => <LayersContainer groups={[[]]} openComments={null} />;

DefaultView.story = {
  name: 'default'
};
