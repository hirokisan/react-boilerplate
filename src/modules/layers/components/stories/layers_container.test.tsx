/* eslint-disable react/display-name */

import React from 'react';

jest.mock('../layers_tabs', () => ({
  LayersTabs: () => <div>Tabs</div>
}));

jest.mock('../tools_container', () => ({
  ToolsContainer: () => <div>Tools</div>
}));

import renderer from 'react-test-renderer';

import { DefaultView } from './layers_container.stories';

describe('Tools View', () => {
  const tree = renderer.create(DefaultView());

  it('renders default', () => {
    expect(tree).toMatchSnapshot();
  });
});
