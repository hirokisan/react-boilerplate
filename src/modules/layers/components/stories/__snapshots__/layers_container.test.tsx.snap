// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tools View renders default 1`] = `
<div
  className="css-1i42z24-LayerContainer"
>
  <div>
    Tabs
  </div>
  <div>
    Tools
  </div>
</div>
`;
